<details>
<summary>Pre-Commit Test failed! Click here</summary>

```
[INFO] Initializing environment for git://github.com/Lucas-C/pre-commit-hooks.
[INFO] Initializing environment for git://github.com/pre-commit/pre-commit-hooks.
[INFO] Initializing environment for git://github.com/pycqa/pydocstyle.git.
[INFO] Initializing environment for https://github.com/pre-commit/pre-commit-hooks.
[INFO] Initializing environment for https://github.com/pre-commit/mirrors-mypy.
[INFO] Initializing environment for https://github.com/psf/black.
[INFO] Initializing environment for https://gitlab.com/PyCQA/flake8.
[INFO] Initializing environment for https://gitlab.com/PyCQA/flake8:pep8-naming.
[INFO] Installing environment for git://github.com/Lucas-C/pre-commit-hooks.
[INFO] Once installed this environment will be reused.
[INFO] This may take a few minutes...
[INFO] Installing environment for git://github.com/pre-commit/pre-commit-hooks.
[INFO] Once installed this environment will be reused.
[INFO] This may take a few minutes...
[INFO] Installing environment for git://github.com/pycqa/pydocstyle.git.
[INFO] Once installed this environment will be reused.
[INFO] This may take a few minutes...
[INFO] Installing environment for https://github.com/pre-commit/pre-commit-hooks.
[INFO] Once installed this environment will be reused.
[INFO] This may take a few minutes...
[INFO] Installing environment for https://github.com/pre-commit/mirrors-mypy.
[INFO] Once installed this environment will be reused.
[INFO] This may take a few minutes...
[INFO] Installing environment for https://github.com/psf/black.
[INFO] Once installed this environment will be reused.
[INFO] This may take a few minutes...
[INFO] Installing environment for https://gitlab.com/PyCQA/flake8.
[INFO] Once installed this environment will be reused.
[INFO] This may take a few minutes...
Tabs remover.............................................................Passed
Trim Trailing Whitespace.................................................Passed
Check for merge conflicts................................................Passed
Fix End of Files.........................................................Passed
Tests should end in _test.py.........................(no files to check)Skipped
Check for added large files..............................................Passed
Check for byte-order marker..............................................Passed
Check for case conflicts.................................................Passed
Check docstring is first.................................................Passed
Check JSON...............................................................Passed
Check for broken symlinks............................(no files to check)Skipped
Detect Private Key.......................................................Passed
Check python ast.........................................................Passed
Debug Statements (Python)................................................Passed
pydocstyle...............................................................Passed
Check Toml...............................................................Passed
Check Yaml...............................................................Passed
Fix End of Files.........................................................Passed
Trim Trailing Whitespace.................................................Passed
mypy.....................................................................Passed
black....................................................................Passed
flake8...................................................................Failed
- hook id: flake8
- exit code: 1

consumer.py:28:1: F401 'thoth.advise_reporter.metrics.prometheus_registry' imported but unused
```
</details>

--- a/consumer.py
+++ b/consumer.py
@@ -18,13 +18,13 @@
 
 """Consume messages produced by advise-reporter.py faust app."""
 
+import os
 import logging
 
 from thoth.advise_reporter import __service_version__
 
 from thoth.messaging import MessageBase, AdviseJustificationMessage
 
-from thoth.advise_reporter.metrics import prometheus_registry
 from thoth.advise_reporter.advise_reporter import expose_metrics
 
 from aiohttp import web
@@ -49,7 +49,7 @@ advise_justification_topic = AdviseJustificationMessage().topic
 @app.page("/metrics")
 async def get_metrics(self, request):
     """Serve the metrics from the consumer registry."""
-    return web.Response(text=generate_latest(prometheus_registry).decode("utf-8"))
+    return web.Response(text=generate_latest().decode("utf-8"))
 
 
 @app.page("/_health")

--- a/thoth/advise_reporter/advise_reporter.py
+++ b/thoth/advise_reporter/advise_reporter.py
@@ -54,9 +54,9 @@ def retrieve_adviser_reports_justifications(adviser_version: str):
 @metrics.in_progress.track_inprogress()
 def expose_metrics(advise_justification: MessageBase):
     """Retrieve adviser reports justifications."""
-    metrics.advise_justification_type.labels(
+    metrics.advise_justification_type_number.labels(
         advise_message=advise_justification.message, thoth_environment=_THOTH_ENVIRONMENT
     ).set(advise_justification.count)
-    _LOGGER.info("advise_message_number(%r)=%r", advise_justification.message, advise_justification.count)
+    _LOGGER.info("advise_justification_type_number(%r)=%r", advise_justification.message, advise_justification.count)
 
     metrics.success.inc()

--- a/thoth/advise_reporter/advise_reporter.py
+++ b/thoth/advise_reporter/advise_reporter.py
@@ -56,7 +56,7 @@ def expose_metrics(advise_justification: MessageBase):
     """Retrieve adviser reports justifications."""
     metrics.advise_justification_type_number.labels(
         advise_message=advise_justification.message, thoth_environment=_THOTH_ENVIRONMENT
-    ).set(advise_justification.count)
+    ).inc(advise_justification.count)
     _LOGGER.info("advise_justification_type_number(%r)=%r", advise_justification.message, advise_justification.count)
 
     metrics.success.inc()

--- a/thoth/advise_reporter/advise_reporter.py
+++ b/thoth/advise_reporter/advise_reporter.py
@@ -23,23 +23,12 @@ import os
 from typing import Dict, Any
 from thoth.lab import adviser
 from thoth.messaging import MessageBase
-<<<<<<< HEAD
-from prometheus_client import CollectorRegistry, Gauge, push_to_gateway
-=======
-from thoth.advise_reporter.metrics import METRIC_ADVISE_TYPE
->>>>>>> Add metrics
+from thoth.advise_reporter import metrics
 
 _LOGGER = logging.getLogger(__name__)
 
 _THOTH_ENVIRONMENT = os.environ["THOTH_DEPLOYMENT_NAME"]
 
-<<<<<<< HEAD
-_THOTH_METRICS_PUSHGATEWAY_URL = os.getenv(
-    "PROMETHEUS_PUSHGATEWAY_URL", "pushgateway-dh-prod-monitoring.cloud.datahub.psi.redhat.com:80"
-)
-
-=======
->>>>>>> Add metrics
 
 def retrieve_adviser_reports_justifications(adviser_version: str):
     """Retrieve adviser reports justifications."""
@@ -61,9 +50,13 @@ def retrieve_adviser_reports_justifications(adviser_version: str):
     return advise_justifications
 
 
+@metrics.exceptions.count_exceptions()
+@metrics.in_progress.track_inprogress()
 def expose_metrics(advise_justification: MessageBase):
     """Retrieve adviser reports justifications."""
-    METRIC_ADVISE_TYPE.labels(advise_message=advise_justification.message, thoth_environment=_THOTH_ENVIRONMENT).set(
-        advise_justification.count
-    )
+    metrics.advise_justification_type.labels(
+        advise_message=advise_justification.message, thoth_environment=_THOTH_ENVIRONMENT
+    ).set(advise_justification.count)
     _LOGGER.info("advise_message_number(%r)=%r", advise_justification.message, advise_justification.count)
+
+    metrics.success.inc()

--- a/thoth/advise_reporter/metrics.py
+++ b/thoth/advise_reporter/metrics.py
@@ -34,8 +34,8 @@ exceptions = Counter("investigator_exceptions", "Number of investigation message
 success = Counter("investigators_processed", "Number of investigation messages which were successfully processed.")
 
 # Advise justifications
-advise_justification_type = Gauge(
-    "thoth_advise_message_number",
-    "Number of thamos advise provided per message.",
+advise_justification_type_number = Gauge(
+    "thoth_advise_justification_type_number",
+    "Number of advise justification provided per justification type.",
     ["advise_message", "thoth_environment"],
 )

--- a/thoth/advise_reporter/metrics.py
+++ b/thoth/advise_reporter/metrics.py
@@ -21,25 +21,20 @@
 
 from thoth.advise_reporter import __service_version__
 
-from prometheus_client import CollectorRegistry, Gauge, Counter
+from prometheus_client import Gauge, Counter
 
-
-# A registry for the consumer metrics...
-prometheus_registry = CollectorRegistry()
 
 # add the application version info metric
-_API_GAUGE_METRIC = Gauge(
-    "advise_reporter_consumer_info",
-    "Adviser reporter Version Info",
-    labelnames=["version"],
-    registry=prometheus_registry,
-)
-_API_GAUGE_METRIC.labels(version=__service_version__).inc()
+advise_reporter_info = Gauge("advise_reporter_consumer_info", "Adviser reporter Version Info", labelnames=["version"])
+advise_reporter_info.labels(version=__service_version__).inc()
 
+# Metrics for Kafka
+in_progress = Gauge("investigators_in_progress", "Total number of investigation messages currently being processed.")
+exceptions = Counter("investigator_exceptions", "Number of investigation messages which failed to be processed.")
+success = Counter("investigators_processed", "Number of investigation messages which were successfully processed.")
 
 # Advise justifications
-
-METRIC_ADVISE_TYPE = Gauge(
+advise_justification_type = Gauge(
     "thoth_advise_message_number",
     "Number of thamos advise provided per message.",
     ["advise_message", "thoth_environment"],

